File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--retrieve-the-sharepoint-project-service_5.cs
            Microsoft.VisualStudio.Shell.ServiceProvider serviceProvider =
                new Microsoft.VisualStudio.Shell.ServiceProvider(
                _applicationObject as Microsoft.VisualStudio.OLE.Interop.IServiceProvider);

            Microsoft.VisualStudio.SharePoint.ISharePointProjectService projectService = 
                serviceProvider.GetService(typeof(Microsoft.VisualStudio.SharePoint.ISharePointProjectService))
                as Microsoft.VisualStudio.SharePoint.ISharePointProjectService;

            if (projectService != null)
            {
                projectService.Logger.WriteLine("This message was written by using the SharePoint project service.",
                    Microsoft.VisualStudio.SharePoint.LogCategory.Message);
            }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--retrieve-the-sharepoint-project-service_5.vb
        Dim serviceProvider As Microsoft.VisualStudio.Shell.ServiceProvider = _
            New Microsoft.VisualStudio.Shell.ServiceProvider( _
                TryCast(_applicationObject, Microsoft.VisualStudio.OLE.Interop.IServiceProvider))

        Dim projectService As Microsoft.VisualStudio.SharePoint.ISharePointProjectService = _
            TryCast(serviceProvider.GetService(GetType(Microsoft.VisualStudio.SharePoint.ISharePointProjectService)), _
                Microsoft.VisualStudio.SharePoint.ISharePointProjectService)

        If projectService IsNot Nothing Then
            projectService.Logger.WriteLine("This message was written by using the SharePoint project service.", _
                Microsoft.VisualStudio.SharePoint.LogCategory.Message)
        End If